### PR TITLE
Update c-interop-native-extensions.md

### PR DESCRIPTION
**Pull request by mraleph**
* Remove the text in line with removal of the feature itself.
* Clarify that embedding Dart VM is still supported and link to a good example of that.

### CODE REVIEW

File: src/server/c-interop-native-extensions.md [hunk 1]
```diff
@@ -3,535 +3,20 @@ title: Native extensions for the standalone Dart VM
...
 
-_Written by William Hesse <br>
-May 2012_
-
 {{site.alert.note}}
-The extension mechanism discussed in this page is for deep integration of the VM
-by custom Dart embedders.
-This extension mechanism is **unsupported** for extension purposes,
-and is **[being removed](https://github.com/dart-lang/sdk/issues/45451)**.
+The extension mechanism which was previously discussed on this page has been
```
Comment (mit-mit) [inline src/server/c-interop-native-extensions.md:7]: Maybe `...previously discussed on this page -- native extensions -- has been` ?

That's likely better for SEO

File: src/server/c-interop-native-extensions.md [hunk 2]
```diff
@@ -3,535 +3,20 @@ title: Native extensions for the standalone Dart VM
...
-
-### Building on Windows
-
-The Windows DLL compilation is complicated by the fact that we need to link with
-library file, dart.lib, that does not contain code itself, but specifies that
-calls to the Dart Embedding API will be resolved by linking to the Dart
-executable, dart.exe, when the DLL is dynamically loaded. This library file is
-generated when building dart and is included in the Dart SDK.
+This deprecation does not affect use-cases where Dart VM is embedded as library
+into another application. Dart Embedding API (as defined in
```
Comment (mit-mit) [inline src/server/c-interop-native-extensions.md:14]: s/Dart Embedding/The Dart Embedding/

File: src/server/c-interop-native-extensions.md [hunk 3]
```diff
@@ -3,535 +3,20 @@ title: Native extensions for the standalone Dart VM
...
+[`include/dart_api.h`] header) remains supported for such use cases. Dart team
```
Comment (mit-mit) [inline src/server/c-interop-native-extensions.md:15]: s/Dart team/The Dart team/

File: src/server/c-interop-native-extensions.md [hunk 4]
```diff
@@ -3,535 +3,20 @@ title: Native extensions for the standalone Dart VM
...
+currently does not provide any guides on how to embed Dart VM into you
```
Comment (mit-mit) [inline src/server/c-interop-native-extensions.md:16]: s/Dart VM/the Dart VM/

Comment (mit-mit) [inline src/server/c-interop-native-extensions.md:16]: s/you/your/

File: src/server/c-interop-native-extensions.md [hunk 5]
```diff
@@ -3,535 +3,20 @@ title: Native extensions for the standalone Dart VM
...
+application, however there are [examples] maintained by the community which
+demonstrate necessary glue code.
```
Comment (mit-mit) [inline src/server/c-interop-native-extensions.md:18]: s/necessary/the necessary/

**Review by mit-mit — CHANGES_REQUESTED** [review]

**Comment by mraleph** [comment]
Comments addressed. PTAL

**Review by mit-mit — APPROVED** [review]

File: src/server/c-interop-native-extensions.md [hunk 6]
```diff
@@ -3,535 +3,22 @@ title: Native extensions for the standalone Dart VM
...
 
-_Written by William Hesse <br>
-May 2012_
-
 {{site.alert.note}}
-The extension mechanism discussed in this page is for deep integration of the VM
-by custom Dart embedders.
-This extension mechanism is **unsupported** for extension purposes,
-and is **[being removed](https://github.com/dart-lang/sdk/issues/45451)**.
+previously discussed on this page  has been
```
Comment (parlough) [inline src/server/c-interop-native-extensions.md:7]:
```suggestion
```

File: src/server/c-interop-native-extensions.md [hunk 7]
```diff
@@ -3,535 +3,22 @@ title: Native extensions for the standalone Dart VM
...
-library file, dart.lib, that does not contain code itself, but specifies that
-calls to the Dart Embedding API will be resolved by linking to the Dart
-executable, dart.exe, when the DLL is dynamically loaded. This library file is
-generated when building dart and is included in the Dart SDK.
+This deprecation does not affect use-cases where Dart VM is embedded as library
+into another application. The Dart Embedding API (as defined in
+[`include/dart_api.h`] header) remains supported for such use cases. The Dart
+team currently does not provide any guides on how to embed the Dart VM into your
+application, however there are [examples] maintained by the community which
+demonstrate the necessary glue code.
```
Comment (parlough) [inline src/server/c-interop-native-extensions.md:20]: Adds some semantic splits and fixes some minor grammatical/style errors.

```suggestion
This deprecation does not affect use-cases where the Dart VM 
is embedded as a library into another application. 
The Dart Embedding API (as defined in the [`include/dart_api.h`][] header) 
remains supported for such use cases. 
The Dart team currently does not provide any guidelines
on how to embed the Dart VM into your application, 
however there are [examples][] maintained by the community
which demonstrate the necessary glue code.
```

**Review by parlough — COMMENTED** [review]
Some minor suggestions and fixes.

**Comment by mraleph** [comment]
@parlough thanks!

**Comment by mraleph** [comment]
@mit-mit @parlough could you merge if you have the commit rights? I apparently don't 🤔

**Pull request merged**